wsl-pro-service/internal/daemon/daemon.go

```diff
@@ -301,6 +309,33 @@ func (d *Daemon) connect(ctx context.Context) (server *streams.Server, err error
 	return streams.NewServer(ctx, d.system, conn), nil
 }
 
+// newTLSConfigFromDir loads certificates from the provided certs path and returns a matching tls.Config.
+func newTLSConfigFromDir(certsPath string) (conf *tls.Config, err error) {
+	decorate.OnError(&err, "could not load TLS config")
+
+	cert, err := tls.LoadX509KeyPair(filepath.Join(certsPath, "client_cert.pem"), filepath.Join(certsPath, "client_key.pem"))
```

Comment:
I notice that the filename is hardcoded more than in one place, this should probably go in common so that those reused name across multiple projects align on what they fetch? (also, as said in a previous comment, this should be more explicit, like wslinstance_cert.pem or any better name)

Reply:
> also, as said in a previous comment, this should be more explicit, like wslinstance_cert.pem or any better name

Let me interject on that part: the same client certificates are used by all WSL instances and the GUI, that's why the generic name. I don't think it's worth creating an exclusive pair of certificates for the GUI, it's doable, but unnecessary in my opinion.

Moving to common as constants and adding comments will clarify that more.